src/cascadia/WindowsTerminal/TrayIcon.cpp

```diff
@@ -0,0 +1,227 @@
+                }
+
+                AppendMenu(submenu, MF_STRING, gsl::narrow<UINT_PTR>(id), displayText.c_str());
+            }
+
+            MENUINFO submenuInfo{};
+            submenuInfo.cbSize = sizeof(MENUINFO);
+            submenuInfo.fMask = MIM_MENUDATA;
+            submenuInfo.dwStyle = MNS_NOTIFYBYPOS;
+            submenuInfo.dwMenuData = (UINT_PTR)TrayMenuItemAction::SummonWindow;
+            SetMenuInfo(submenu, &submenuInfo);
+
+            AppendMenu(hmenu, MF_POPUP, (UINT_PTR)submenu, RS_(L"TrayIconWindowSubmenu").c_str());
+        }
+    }
+    return hmenu;
+}
+
+// Method Description:
+// - This is the handler for when one of the menu items are selected within
+//   the tray icon's context menu.
+// Arguments:
+// - menu: The handle to the menu that holds the menu item that was selected.
+// - menuItemIndex: The index of the menu item within the given menu.
+// Return Value:
+// - <none>
+void TrayIcon::TrayMenuItemSelected(const HMENU menu, const UINT menuItemIndex)
+{
+    // Check the menu's data for a specific action.
+    MENUINFO mi{};
+    mi.cbSize = sizeof(MENUINFO);
+    mi.fMask = MIM_MENUDATA;
+    GetMenuInfo(menu, &mi);
+    if (mi.dwMenuData)
+    {
+        if (gsl::narrow<TrayMenuItemAction>(mi.dwMenuData) == TrayMenuItemAction::SummonWindow)
+        {
+            winrt::Microsoft::Terminal::Remoting::SummonWindowSelectionArgs args{};
+            args.WindowID(GetMenuItemID(menu, menuItemIndex));
+            args.SummonBehavior().ToggleVisibility(false);
+            _SummonWindowRequestedHandlers(args);
+            return;
+        }
+    }
+
+    // Now check the menu item itself for an action.
+    const auto action = gsl::narrow<TrayMenuItemAction>(GetMenuItemID(menu, menuItemIndex));
+    switch (action)
+    {
+    case TrayMenuItemAction::FocusTerminal:
+    {
+        winrt::Microsoft::Terminal::Remoting::SummonWindowSelectionArgs args{};
+        args.SummonBehavior().ToggleVisibility(false);
+        _SummonWindowRequestedHandlers(args);
+        break;
+    }
+    }
+}
+
+// Method Description:
+// - This is the handler for when the tray icon itself is left-clicked.
+// Arguments:
+// - <none>
+// Return Value:
+// - <none>
+void TrayIcon::TrayIconPressed()
+{
+    // No name in the args means summon the mru window.
+    winrt::Microsoft::Terminal::Remoting::SummonWindowSelectionArgs args{};
+    args.SummonBehavior().ToggleVisibility(false);
+    _SummonWindowRequestedHandlers(args);
+}
+
+// Method Description:
+// - Re-add a tray icon using our currently saved tray icon data.
+// Arguments:
+// - <none>
+// Return Value:
+// - <none>
+void TrayIcon::ReAddTrayIcon()
+{
+    Shell_NotifyIcon(NIM_ADD, &_trayIconData);
+    Shell_NotifyIcon(NIM_SETVERSION, &_trayIconData);
+}
+
+// Method Description:
+// - Deletes our tray icon.
+// Arguments:
+// - <none>
+// Return Value:
+// - <none>
+void TrayIcon::DestroyTrayIcon()
+{
+    Shell_NotifyIcon(NIM_DELETE, &_trayIconData);

```

Comment:
I was intending for this function to be more of a `RemoveIconFromTray` instead of a "destroy" (I'll rename). I wanted the actual "destruction" of the tray icon  to be when `AppHost` sets its `_trayIcon` ptr to null, and if it ever wanted to get a new tray icon, it should instantiate a new one.